### ISSUE TITLE
Inactive deployments in OSB domain

### ISSUE THREAD

**Issue by pwninja** [oracle/weblogic-deploy-tooling#919]
I am installing an OSB domain using WDT 1.9.14 and it is provisioned with 168 deployments, but some of them are in status Installed. How to make them Active?
Here is the list - they are mostly in Scope of "OSBBothTemplate template" but some are "Global":

Aggregator Singleton Marker Application
beehive-controls-1.0.1-10.0-war (1.0,1.0.2.2)
beehive-netui-1.0.1-10.0 (1.0,1.0.2.2)
beehive-netui-resources-1.0.1-10.0 (1.0,1.0.2.2)
CoherenceAdapter
com.bea.wlp.lwpf.console.app (10.3.0,10.3.0)
com.bea.wlp.lwpf.console.web (10.3.0,10.3.0)
JCAFrameworkImpl (12.1.2.0,12.1.2.0)
JDEWorldAdapter
jstl (1.2,1.2.0.1)
LdapAdapter
MQSeriesAdapter
MSMQAdapter
oracle.soa.commonconsole.dependencies (12.1.2,12.1.2)
oracle.soa.sb.em.adf.mgmt (1.0,12.1.2.0.0)
oracle.soa.webmapper (12.1.3,12.1.3)
OracleAppsAdapter
OracleBamAdapter
resource-mq-connection (12.1.3,12.1.3)
resource-mq-jms-connector (12.1.3,12.1.3)
SAPAdapter
SiebelAdapter
SocketAdapter
stage-logging (12.1.3,12.1.3)
stage-publish (12.1.3,12.1.3)
stage-routing (12.1.3,12.1.3)
stage-transform (12.1.3,12.1.3)
stage-utils (12.1.3,12.1.3)
struts-1.2 (1.2,1.2.9)
transport-dsp (12.1.3,12.1.3)
transport-ejb (12.1.3,12.1.3)
transport-email (12.1.3,12.1.3)
transport-file (12.1.3,12.1.3)
transport-ftp (12.1.3,12.1.3)
transport-jca (12.1.3,12.1.3)
transport-jejb (12.1.3,12.1.3)
transport-mq (12.1.3,12.1.3)
transport-pollersdk (12.1.3,12.1.3)
transport-sftp (12.1.3,12.1.3)
transport-soa (12.1.3,12.1.3)
transport-tuxedo (12.1.3,12.1.3)
UMSAdapter
weblogic-controls-10.0-war (10.0,10.2)
wlp-framework-common-web-lib (10.3.0,10.3.0)
wlp-framework-struts-1.2-web-lib (10.3.0,10.3.0)
wlp-light-web-lib (10.3.0,10.3.0)
wlp-lookandfeel-web-lib (10.3.0,10.3.0)
wls-commonslogging-bridge-war (1.0,1.1)

I tried defining some of those by explicitly putting into WDT model file, like this:
```
appDeployments:
    Library:
        'transport-mq#12.1.3@12.1.3':
            CompatibilityName: 'transport-mq#12.1.3@12.1.3'
            SourcePath: '@@ORACLE_HOME@@/osb/lib/transports/mq-transport.jar'
            DeploymentOrder: 44
            StagingMode: nostage
            Target: 'OsbCluster,AdminServer'
```
but it didn't work - deployment stays in Installed state, target is not getting set. Not to mention that this seems to be redundant to define it, if it is already deployed.

**Comment by pwninja** [comment]
I have that one observation - with these 3 Libraries as an example:

- beehive-controls-1.0.1-10.0-war (1.0,1.0.2.2)
- beehive-netui-1.0.1-10.0 (1.0,1.0.2.2)
- beehive-netui-resources-1.0.1-10.0 (1.0,1.0.2.2)

Once domain is installed they are visible in Deployments tab with no Target and Scope set to OSBBothTemplate. When I removed Resource Group Templates (including OSBBothTemplate), the Target for these libraries automatically changed to "AdminServer, OsbCluster" and the Scope changed to Global.
Removing Resource Group Templates is a subject of automation that we want to apply after installing the domain (compare: #853 ), but still I wonder if that's the proper way to handle target and state of Deployments.

**Comment by CarolynRountree** [comment]
Can you try something, and target these three libraries. When removing the resource group template, WLS might want to force it to a target and if it is already targeted, might not target admin server.

**Comment by pwninja** [comment]
I am not sure the exact steps of this test - in my WDT model I have for example mq-transport Library defined - while removing ResourceGroupTemplates it also got Active, got a target and scope Global. So do you want me to assign it a target different then "AdminServer, OsbCluster" and see which one comes up?

**Comment by pwninja** [comment]
I made it this way:
```
appDeployments:
    Library:
        'transport-mq#12.1.3@12.1.3':
            CompatibilityName: 'transport-mq#12.1.3@12.1.3'
            SourcePath: '@@ORACLE_HOME@@/osb/lib/transports/mq-transport.jar'
            DeploymentOrder: 44
            StagingMode: nostage
            Target: 'OsbCluster'
        'transport-sftp':
            CompatibilityName: 'transport-sftp#12.1.3@12.1.3'
            SourcePath: '@@ORACLE_HOME@@/osb/lib/transports/sftp-transport.jar'
            DeploymentOrder: 44
            StagingMode: nostage
            Target: 'OsbCluster,!AdminServer'
```
After deployment both libraries were in Scope od OSBBothTemplate.
I deleted ResourceGroupTemplates and in result libraries got activated and the target was:

- transport-mq: AdminServer, OsbCluster
- transport-sftp: OsbCluster

**Comment by CarolynRountree** [comment]
To clarify, when you explicitly target, the resources are deployed and activated - else they are in install state?
So the challenge is the number of resources that would have to be explicitly targeted?

**Comment by pwninja** [comment]
This is what I included in the initial ticket description - when I explicitly target the libraries they come with:
* Status: Installed
* Target: empty
* Scope: OSBBothTemplate

It is only the removal of ResourceGroupTemplates that I found to activate the libraries.

**Comment by ddsharpe** [comment]
@pwninja I had a long email conversation with the SOA PM and Dev team.  The conclusion was that they are not ready to support WDT.  SOA and OSB would be nice to have in WDT, but the WLS/WDT team does not have the background to determine what WDT should or should not do for SOA/OSB domains.

**Comment by CarolynRountree** [comment]
OSB provided a fix ARU 24175240 for this issue. 

```
Pls check : https://bug.oraclecorp.com/pls/bug/webbug_edit.edit_info_top?rptno=32343813, this issue is fixed via this bug by WLS team, even without the fix you can delete the ResouceGroupTemplate without any issue.

Just to add, it Is only a display issue in WLS console, the apps are deployed with Global scope, but are  visible in the console
```

**Comment by pwninja** [comment]
Applying the patch 32343813 "DEPLOYMENT SCOPE IN WLS CONSOLE IS WRONG" fixed the problem. Thanks!
